components/modals/DeleteProjectModal/DeleteProjectModal.tsx

```diff
@@ -0,0 +1,88 @@
+import useApiCall from "@/hooks/useApiCall";
+import { Mutate } from "@/hooks/useFetch";
+import { HTTP_METHOD } from "@/types/api";
+import { Project } from "@/types/projects";
+import { Stack, Button } from "@mui/material";
+import { Dispatch, FC, SetStateAction, useState } from "react";
+import Modal from "../Modal";
+
+type Props = {
+  project: Project;
+  open: boolean;
+  setOpen: Dispatch<SetStateAction<boolean>>;
+  mutate: Mutate<Project[]>;
+  setSuccess: (message: string) => void;
+  setError: (error: unknown) => void;
+};
+
+const DeleteProjectModal: FC<Props> = ({
+  project,
+  open,
+  setOpen,
+  mutate,
+  setSuccess,
+  setError,
+}) => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const deleteProject = useApiCall({
+    method: HTTP_METHOD.DELETE,
+    endpoint: `/projects/${project.id}`,
+    onSuccess: () => {
+      mutate((projects) => {
+        const deletedProjectId = project.id;
+        const deletedProjectIdx = projects.findIndex(
+          (project) => project.id === deletedProjectId
+        );
+        const newProjects = [...projects];
+        newProjects.splice(deletedProjectIdx, 1);
+        return newProjects;
+      });
+    },
+  });
+
+  const handleDelete = async () => {
+    setIsSubmitting(true);
+    try {
+      await deleteProject.call();
+      setSuccess(`You have successfully deleted the project ${project.name}!`);
+      handleCloseModal();
+    } catch (error) {
+      setError(error);
+    }
+    setIsSubmitting(false);
+  };
+
+  const handleCloseModal = () => {
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <Modal
+        open={open}
+        handleClose={handleCloseModal}
+        title={`Delete Project`}
+        subheader={`You are deleting project ${project.name}. This action is irreversible, are you sure?`}
+        sx={{ width: "400px" }}
+      >
+        <Stack direction="row" justifyContent="space-between">
+          <Button size="small" onClick={handleCloseModal}>
+            Cancel
+          </Button>
+          <Button
+            size="small"
+            variant="contained"
+            color="error"
+            onClick={handleDelete}
+            disabled={isSubmitting}
```

Comment:
Can remove the `isSubmitting` useState and just use this
```suggestion
            disabled={isCalling(deleteProject.status)}
```